website/client/js/controllers/settingsCtrl.js

```diff
@@ -287,6 +293,33 @@ habitrpg.controller('SettingsCtrl',
       return Math.floor(numberOfHourglasses);
     };
 
+    $scope.hasSocialAuth = function(user) {
+      SOCIAL_AUTH_NETWORKS.forEach(function (network) {
+        if (user.auth.hasOwnProperty(network.key)) {
+          return Boolean(user.auth[network.key].id)
```

Comment:
You should use `find` not `forEach`

```js
$scope.hasSocialAuth = function (user) {
  return SOCIAL_AUTH_NETWORKS.find(function (network) {
    if (user.auth.hasOwnProperty(network.key)) {
      return user.auth[network.key].id;
    }
  })
};
```

This will return true if it can find the social auth and return undefined if it cannot.

The way you have it, this method will _always_ return false because `forEach`s don't break like `for loop`s do.